### PR TITLE
multiview component

### PR DESCRIPTION
**Pull request by TrevorDev**
https://github.com/BabylonJS/Babylon.js/issues/6015

### CODE REVIEW

File: src/scene.ts [hunk 1]
```diff
@@ -1046,7 +1047,8 @@ export class Scene extends AbstractScene implements IAnimatable {
     public proceduralTexturesEnabled = true;
 
     // Private
-    private _engine: Engine;
+    /** @hidden */
+    public _engine: Engine;
```
Comment (sebavan) [inline src/scene.ts]: Why not using getEngine() when you need engine ?

File: src/scene.ts [hunk 2]
```diff
@@ -870,7 +870,8 @@ export class Scene extends AbstractScene implements IAnimatable {
     /** All of the active cameras added to this scene. */
     public activeCameras = new Array<Camera>();
 
-    private _activeCamera: Nullable<Camera>;
+    /** @hidden */
+    public _activeCamera: Nullable<Camera>;
```
Comment (sebavan) [inline src/scene.ts]: Why not relying on getActiveCamera()

File: src/scene.ts [hunk 3]
```diff
@@ -1112,18 +1114,19 @@ export class Scene extends AbstractScene implements IAnimatable {
     /** @hidden */
     public _activeAnimatables = new Array<Animatable>();
 
-    private _transformMatrix = Matrix.Zero();
-    private _transformMatrixR = Matrix.Zero();
+    /** @hidden */
+    public _transformMatrix = Matrix.Zero();
```
Comment (sebavan) [inline src/scene.ts]: Not sure if the previous comment applies for matrices as well ?

File: src/scene.ts [hunk 4]
```diff
@@ -2585,16 +2578,8 @@ export class Scene extends AbstractScene implements IAnimatable {
             Frustum.GetPlanesToRef(this._transformMatrix, this._frustumPlanes);
         }
 
-        if (viewR && projectionR) {
-            viewR.multiplyToRef(projectionR, Tmp.Matrix[0]);
-            Frustum.GetRightPlaneToRef(Tmp.Matrix[0], this._frustumPlanes[3]); // Replace right plane by second camera right plane
-        }
-
         if (this._multiviewSceneUbo && this._multiviewSceneUbo.useUbo) {
```
Comment (sebavan) [inline src/scene.ts]: Could this be done by callbacks like the other components instead of the test ?

File: src/scene.ts [hunk 5]
```diff
@@ -4169,32 +4155,7 @@ export class Scene extends AbstractScene implements IAnimatable {
         }
 
         if (camera._useMultiviewToSingleView) {
```
Comment (sebavan) [inline src/scene.ts]: Same here could it be done by the component callbacks ?

File: src/scene.ts (hunk 2 above)
Comment (TrevorDev) [reply src/scene.ts]: In the component I am doing this._activeCamera = camera._rigCameras[index]; if I use activeCamera instead, a camera changed notification occurs which shouldnt happen when it changes just for rendering.

Comment (sebavan) [reply src/scene.ts]: Makes total sense here 👍

File: src/scene.ts (hunk 4 above)
Comment (TrevorDev) [reply src/scene.ts]: I am finding using the observables a bit tricky as for multiview it's not really doing something extra, it is doing something different instead. I could add something like a renderMultiviewToSingleViewObservable but then it's not any different than a function call.

**Comment by sebavan** [comment]
can not reply in the conversation for some reasons. Understood for the callbacks, do not bother then

**Pull request merged**